packages/providers/src/AppRouter.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { createContext, ReactNode, useContext, useMemo } from 'react';
+import { RouterProvider } from 'react-aria-components';
+import { FlattenToAppURLProvider } from '@plone/components';
+import { flattenToAppURL } from './utils';
+
+export type AnySearchSchema = {};
+export interface Location<TSearchObj extends AnySearchSchema = {}> {
+  href: string;
+  pathname: string;
+  search: TSearchObj;
+  searchStr: string;
+  hash: string;
+}
+
+interface AppRouter {
+  useLocation: () => Location | undefined;
+  navigate: (path: string) => void;
+  useHref: (to: string, options?: any) => string;
+  flattenToAppURL: (path: string | undefined) => string | undefined;
+}
+
+const AppRouterContext = createContext<AppRouter>({
+  useLocation: () => ({
+    href: '',
+    pathname: '',
+    search: {},
+    searchStr: '',
+    hash: '',
+  }),
+  navigate: () => {},
+  useHref: () => '',
+  flattenToAppURL,
+});
+
+interface AppRouterProps {
+  useLocation: () => Location | undefined;
+  navigate: (path: string) => void;
+  useHref?: (to: string, options?: any) => string;
+  flattenToAppURL: (path: string | undefined) => string | undefined;
+  children: ReactNode;
+}
+
+export function AppRouterProvider(props: AppRouterProps) {
+  let { children, navigate, useLocation, useHref, flattenToAppURL } = props;
+
+  let ctx = useMemo(
+    () => ({
+      useLocation,
+      navigate,
+      useHref,
+      flattenToAppURL,
+    }),
+    [useLocation, navigate, useHref, flattenToAppURL],
+  );
+
+  return (
+    <AppRouterContext.Provider value={ctx}>
+      <RouterProvider navigate={navigate} useHref={useHref}>
+        <FlattenToAppURLProvider flattenToAppURL={flattenToAppURL}>
+          {children}
+        </FlattenToAppURLProvider>
+      </RouterProvider>
+    </AppRouterContext.Provider>
+  );
+}
+
+export function useAppRouter() {
+  const { useLocation } = useContext(AppRouterContext);
+  return useLocation();
+}
```

Comment:
Why only returning the location? What about the other things set in this context?

Reply:
I have to port the fix from the slots PR

Reply:
@pnicolli https://github.com/plone/volto/pull/6069/commits/905de9ef81738fd8c4d9a69e714074e73a3b09c6